packages/itwinui-react/src/types/react-table-config.ts

```diff
@@ -95,8 +95,11 @@ declare module 'react-table' {
       UseSortByOptions<D> {
     /**
      * List of columns.
+     *
+     * Should not have a top-level `Header` or a `columns` sub-property. They are allowed to be passed for backwards compatibility but are not officially supported.
+     * See https://github.com/iTwin/iTwinUI/wiki/iTwinUI-react-v2-migration-guide#breaking-changes`
      */
-    columns: Array<Column<any>>; // eslint-disable-line @typescript-eslint/no-explicit-any
+    columns: Array<Column<D>>; // eslint-disable-line @typescript-eslint/no-explicit-any
```

Comment:
No need to disable eslint anymore

```suggestion
    columns: Array<Column<D>>;
```

Reply:
hmm, yes i will remove. but now i actually can't remember why we made it `any` in the first place. need to investigate

Reply:
Found this PR but the description is not helpful :/
https://github.com/iTwin/iTwinUI-react/pull/940